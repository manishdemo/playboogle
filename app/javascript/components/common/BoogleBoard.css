.square {
    background: #95bb76;
    border: 1px solid #999;
    float: right;
    font-size: 40px;
    font-weight: bold;
    line-height: 50px;
    height: 50px;
    margin-right: -1px;
    margin-top: -1px;
    padding: 0;
    text-align: center;
    width: 50px;
}
.square:focus {
    outline: none;
}

.board-row {
    alignment: center;
    width: 80%;
}

.board-row:after {
    clear: both;
    content: "";
    display: table;
}

.boogle-board {
    alignment: center;
    width: 80%;
}