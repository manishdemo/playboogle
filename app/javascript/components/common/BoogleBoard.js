import React from "react"
import "./BoogleBoard.css"
import * as PropTypes from "prop-types";

class Square extends React.Component {
    render() {
        return (
            <button className="square">
                {this.props.value}
            </button>
        );
    }
}

class BoogleBoard extends React.Component {

    renderSquare(i) {
        return <Square value={this.props.boogleString[i]}  />;
    }

    render() {
        // console.log("In Boogle Board Render")
        let boogleString = this.props.boogleString;
        // console.log(boogleString.charAt(0))
        return (
            <div className="boogle-board">
                <div className="board-row">
                    {this.renderSquare(0)}
                    {this.renderSquare(1)}
                    {this.renderSquare(2)}
                    {this.renderSquare(3)}
                </div>
                <div className="board-row">
                    {this.renderSquare(4)}
                    {this.renderSquare(5)}
                    {this.renderSquare(6)}
                    {this.renderSquare(7)}
                </div>
                <div className="board-row">
                    {this.renderSquare(8)}
                    {this.renderSquare(9)}
                    {this.renderSquare(10)}
                    {this.renderSquare(11)}
                </div>
                <div className="board-row">
                    {this.renderSquare(12)}
                    {this.renderSquare(13)}
                    {this.renderSquare(14)}
                    {this.renderSquare(15)}
                </div>

            </div>
        );
    }

}

BoogleBoard.propTypes = {
    boogleString: PropTypes.string
};

export default BoogleBoard