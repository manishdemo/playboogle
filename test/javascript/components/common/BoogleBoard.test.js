import React from "react";

import BoogleBoard from "../../../../app/javascript/components/common/BoogleBoard";
import {describe, expect, it} from "@jest/globals";
import {mount, shallow} from "enzyme";

describe('BoogleBoard', () => {
    it('should render correctly', () => {
        const component = shallow(<BoogleBoard boogleString="ABCDEFGHIJKLMNOP"/> );
        expect(component).toMatchSnapshot();
    });

    it('should change values in each square as per provided props', () => {
        const props = {
                boogleString: "MANISHBOOGLETEST"
            };
        const boogleBoardWrapper = mount(<BoogleBoard {...props} />);
        expect(boogleBoardWrapper.find('.square').first().text()).toEqual('M');
        expect(boogleBoardWrapper.find('.square').at(1).text()).toEqual('A');
        expect(boogleBoardWrapper.find('.square').at(15).text()).toEqual('T');
        boogleBoardWrapper.unmount();
    });

});