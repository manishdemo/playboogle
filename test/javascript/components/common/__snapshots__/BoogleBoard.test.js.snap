// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoogleBoard should render correctly 1`] = `
<div
  className="boogle-board"
>
  <div
    className="board-row"
  >
    <Square
      value="A"
    />
    <Square
      value="B"
    />
    <Square
      value="C"
    />
    <Square
      value="D"
    />
  </div>
  <div
    className="board-row"
  >
    <Square
      value="E"
    />
    <Square
      value="F"
    />
    <Square
      value="G"
    />
    <Square
      value="H"
    />
  </div>
  <div
    className="board-row"
  >
    <Square
      value="I"
    />
    <Square
      value="J"
    />
    <Square
      value="K"
    />
    <Square
      value="L"
    />
  </div>
  <div
    className="board-row"
  >
    <Square
      value="M"
    />
    <Square
      value="N"
    />
    <Square
      value="O"
    />
    <Square
      value="P"
    />
  </div>
</div>
`;
